src/pykeen/sampling/basic_negative_sampler.py

```diff
@@ -96,15 +96,15 @@ def sample(self, positive_batch: torch.LongTensor) -> Tuple[torch.LongTensor, Op
             # To make sure we don't replace the {head, relation, tail} by the
             # original value we shift all values greater or equal than the original value by one up
             # for that reason we choose the random value from [0, num_{heads, relations, tails} -1]
-            if not self.filtered:
+            if self.filterer is None:
                 negative_batch[start:stop, index] += (
                     negative_batch[start:stop, index] >= positive_batch[start:stop, index]
                 ).long()
 
         # If filtering is activated, all negative triples that are positive in the training dataset will be removed
-        if self.filtered:
-            negative_batch, batch_filter = self.filter_negative_triples(negative_batch=negative_batch)
-        else:
+        if self.filterer is None:
```

Comment:
I'm going to flip the order of this if statement to reduce the diff, unless there's a good reason not to?

Reply:
I just flipped from
```python
if not flag:
  ..
else:
  ..
```
to
```python
if flag:
  ..
else:
 ..
```
since it looked cleaner to me. But smaller diff is also a valid objective :sweat_smile: